boot/zephyr/CMakeLists.txt

```diff
@@ -7,8 +7,12 @@
 cmake_minimum_required(VERSION 3.8.2)
 
 # Board-specific CONF_FILES should get merged into the build as well.
-# Default to qemu_x86 if no board has been specified.
-set(BOARD qemu_x86)
+# If the board is not set by the build system, fail, otherwise the use
+# of BOARD below will mismatch with how this value gets set by the
+# Zephyr build system.
+if(NOT DEFINED BOARD)
+  assert(0 "Must set a BOARD to build on. Please configure a board with -DBOARD=boardname")
```

Comment:
This implies that setting the board through the environment variable BOARD is not supported, but this should also be supported.

Reply:
Not having a default BOARD is sane. But we can't assert like this because there are more ways to set BOARD than -DBOARD=.

Reply:
Doing this would set BOARD correctly:

```
if(NOT BOARD)
        set(BOARD $ENV{BOARD})
endif()

```

One could assert that BOARD is set after this snippet, or one could safely omit the assert and let the assert in boilerplate.cmake enforce that BOARD is set.

Reply:
This still doesn't do the right thing. The only solutions I see are 1. replicate the 30 or so lines in `boilerplate.cmake` that figure out what BOARD should be set to, or 2. Figure out how to move the DTC_OVERLAY setting to after `boilerplate.cmake` is included.

Reply:
Can you describe a situation (reproducable steps) where this does not do the right thing?

We have been using this trick to replicate the lines in boilerplate.cmake to great effect so far.

Reply:
Never mind, assuming Marti's approach (https://github.com/JuulLabs-OSS/mcuboot/pull/577#issuecomment-545640142) works, this point is moot.